woocommerce/woocommerce-ios
# [Mobile Payments] Duplicate CardReaderConnectionController before refactor

Part of : #8080 


### Description


In sTAP Away, we will add a screen at the start of the card reader connection flow for users with supporting hardware to choose whether to connect a bluetooth reader, or use the built in card reader.

To achieve this, we will have separate connection controllers, one for each type of reader which is specialised to connect to a particular type of reader.

This PR duplicates the existing `CardReaderConnectionController`, renames the current one `LegacyCardReaderConnectionController`, and puts us in a position to start refactoring it.

There are no effective code changes in this PR. Apologies for the length, but it is primarily a copy-paste.

### Testing instructions


Check unit tests pass
Check you can take a payment with the feature toggle off, and that `LegacyCardReaderConnectionController.searchAndConnect()` is used there (use a breakpoint to check this)


---
